kernel/src/time/mod.rs

```diff
@@ -86,9 +86,22 @@ impl From<Duration> for timeval_t {
     }
 }
 
-impl From<timeval_t> for Duration {
-    fn from(timeval: timeval_t) -> Self {
-        Duration::new(timeval.sec as u64, (timeval.usec * 1000) as u32)
+impl TryFrom<timeval_t> for Duration {
+    type Error = crate::Error;
+
+    fn try_from(timeval: timeval_t) -> Result<Self> {
+        let sec = timeval.sec;
+        let nsec = timeval.usec * 1000;
```

Comment:
This still has a chance of overflowing, right?

Better to check `sec` and `usec` and then construct `Duration::new(sec as u64, (usec * 1000) as u32)`.

Reply:
Yes, I'm changing the code now, and will push later